retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
Just to clarify: I'm not voting for calling the modes "pushgateway" or "federation" (that's why I put doublequotes around them). We need to find more descriptive names once we have decided to want those modes.

@juliusv As labels applied via the target config are up to the user, we probably still want to apply them in "federation" mode. "auto-attach" is `job` and `instance` so far (but there might be more at some point). Interestingly, one might argue the two modes "federation" and "pushgateway" are doing different things and could even be combined with each other: "Federation" prevents auto-attaching (of `job` and `instance`) while "pushgateway" modifies the way name collisions are handled (which also affects target labels set by the user).

**sigh**